mlir/lib/Dialect/Rock/Transforms/GridLayoutEmitter.cpp

```diff
@@ -35,9 +35,47 @@ using namespace mlir::rock;
 using namespace mlir::arith;
 using namespace mlir::rock::layout;
 
+static Value rearrangeWorkgroupsForXCC(Location loc, PatternRewriter &b,
+                                       Value bid, int64_t gridSize,
+                                       int64_t numChipletsPerGroup) {
+  Value numChipletsVal =
+      b.createOrFold<ConstantIndexOp>(loc, numChipletsPerGroup);
+  int64_t wgsPerChiplet = (gridSize) / numChipletsPerGroup;
+  Value wgsPerChipletVal = b.createOrFold<ConstantIndexOp>(loc, wgsPerChiplet);
+  Value logicalChipletId = b.create<RemUIOp>(loc, bid, numChipletsVal);
+  Value wgIdPerLogicalChiplet = b.create<DivUIOp>(loc, bid, numChipletsVal);
+  Value rearrangedBid = b.create<AddIOp>(
+      loc, wgIdPerLogicalChiplet,
+      b.create<MulIOp>(loc, logicalChipletId, wgsPerChipletVal));
+  int64_t lastNumChipletMultiple =
+      (gridSize - 1) - (gridSize % numChipletsPerGroup);
+  Value lastNumChipletMultipleVal =
+      b.createOrFold<ConstantIndexOp>(loc, lastNumChipletMultiple);
+  Value isBidLargerThanlastNumChipletMultiple = b.create<arith::CmpIOp>(
+      loc, arith::CmpIPredicate::sgt, bid, lastNumChipletMultipleVal);
```

Comment:
This could be `ugt`, maybe? But the arithmetic cleaner will probably do the right thing here

Reply:
ugt gave me verification errors.